integration/aws_upload.go

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2022 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/eclipse-kanto/file-upload/client"
+	"github.com/eclipse-kanto/file-upload/uploaders"
+	"github.com/stretchr/testify/require"
+)
+
+// awsStorageProvider provides testing functionalities for the AWS storage provider
+type awsStorageProvider struct {
+	options       map[string]string
+	uploads       map[string]string
+	client        *s3.Client
+	presignClient *s3.PresignClient
+	t             *testing.T
+}
+
+// NewAWSStorageProvider creates an implementation of the StorageProvide interface for the AWS storage provider,
+// retrieves the needed credentials from environment variables
+func NewAWSStorageProvider(t *testing.T) StorageProvider {
+	creds, err := uploaders.GetAWSTestCredentials()
+	require.NoError(t, err, "AWS credentials not set")
+	options := uploaders.GetAWSTestOptions(creds)
+	options[client.StorageProvider] = uploaders.StorageProviderAWS
+	client, err := uploaders.GetAWSClient(options)
+	require.NoError(t, err, "error getting AWS client")
+	presignClient := s3.NewPresignClient(client)
+	return awsStorageProvider{
+		options:       options,
+		uploads:       make(map[string]string),
+		client:        client,
+		presignClient: presignClient,
+		t:             t,
+	}
+}
+
+func (provider awsStorageProvider) requestUpload(correlationID string, filePath string) map[string]interface{} {
```

Comment:
All these methods are touching the internal struct state i.e. by default, they should work on the reference i.e. a pointer should be used.